assets/javascripts/kitten/components/layout/marger.js

```diff
@@ -1,18 +1,131 @@
-import React from 'react'
+import React, { Component } from 'react'
+import Radium, { StyleRoot } from 'radium'
+import PropTypes from 'prop-types'
+import { ScreenConfig } from 'kitten/constants/screen-config'
 import TYPOGRAPHY from 'kitten/constants/typography-config'
+import isStringANumber from 'is-string-a-number'
+import { upcaseFirst } from 'kitten/helpers/utils/string'
 
-export const Marger = ({ top, bottom, style, ...props }) => {
-  const gutter = 10 / TYPOGRAPHY.root // Transform 10 to .625 for have rem unit.
-  const styles = {
-    ...style,
-    marginTop: top ? `${top * gutter}rem` : null,
-    marginBottom: bottom ? `${bottom * gutter}rem` : null,
+const viewportRanges = ['xxs', 'xs', 's', 'm', 'l', 'xl']
+
+export class MargerBase extends Component {
+  static propTypes = {
+    top: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+      PropTypes.shape({
+        xxs: PropTypes.number,
+        xs: PropTypes.number,
+        s: PropTypes.number,
+        m: PropTypes.number,
+        l: PropTypes.number,
+        xl: PropTypes.number,
+      }),
+    ]),
+    bottom: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+      PropTypes.shape({
+        xxs: PropTypes.number,
+        xs: PropTypes.number,
+        s: PropTypes.number,
+        m: PropTypes.number,
+        l: PropTypes.number,
+        xl: PropTypes.number,
+      }),
+    ]),
+  }
+
+  static defaultProps = {
+    top: null,
+    bottom: null,
+  }
+
+  constructor(props) {
+    super(props)
+
+    // Use 10px as gutter base and transform it to have a rem unit.
+    this.gutter = 10 / TYPOGRAPHY.root
+  }
+
+  marginSize = value => `${value * this.gutter}rem`
+  valueIsNumber = value => {
+    // Retro-compatibility: this handles the case when the user enters `.5` as a
+    // value for a margin.
+    if (typeof value === 'string' && value.match(/^\./)) {
+      return isStringANumber(`0${value}`)
```

Comment:
Est-ce que tout le temps mettre un `0` devant marcherait ? `isStringANumber('042')` va me sortir `42` ?

Reply:
C'est une très bonne question. Je travaille sur `is-string-a-number`, il faut que je check le use-case.

Reply:
Donc, le use-case en ajoutant tout le temps `0` ne va pas fonctionner car `isStringANumber('042') ` renvoie `false` (dû à l'implémentation avec `parseFloat` : https://github.com/Ynote/is-string-a-number/blob/master/index.js).

C'est un très bon cas d'usage, j'ai l'impression que ça devrait renvoyer `true`.

Reply:
Ah mais c'est à toi ! Je me demandais d'où sortait cette lib 👍

Reply:
`parseFloat('042')` sort `42` mais vu que c'est différent de `042`, ca marche pas.

Reply:
Ah oui :D c'est parce que j'ai cherché PARTOUT pour une lib comme ça. Et y'avait rien qui gérait juste des nombres "normaux" (genre pas en hexa ou avec des exposants). Mais, au final, j'en suis déjà à ma 2ème version du module et c'est toujours pas la folie.